$OpenBSD: patch-common_mk,v 1.2 2020/10/30 22:07:20 jeremy Exp $

Enable verbose mode when building.

Don't regenerate rdoc documentation during install.

Force building coroutine code without optimization on sparc64
to avoid crashes.

Index: common.mk
--- common.mk.orig
+++ common.mk
@@ -7,7 +7,7 @@ dll: $(LIBRUBY_SO)
 .SUFFIXES: .rbinc .rb .inc .h .c .y .i .$(ASMEXT) .$(DTRACE_EXT)
 
 # V=0 quiet, V=1 verbose.  other values don't work.
-V = 0
+V = 1
 Q1 = $(V:1=)
 Q = $(Q1:0=@)
 ECHO0 = $(ECHO1:0=echo)
@@ -292,7 +292,7 @@ ext/configure-ext.mk: $(PREP) all-incs $(MKFILES) $(RB
 configure-ext: $(EXTS_MK)
 
 build-ext: $(EXTS_MK)
-	$(Q)$(MAKE) -f $(EXTS_MK) $(mflags) libdir="$(libdir)" LIBRUBY_EXTS=$(LIBRUBY_EXTS) \
+	$(Q)$(MAKE) -f $(EXTS_MK) V=1 $(mflags) libdir="$(libdir)" LIBRUBY_EXTS=$(LIBRUBY_EXTS) \
 	    EXTENCS="$(ENCOBJS)" UPDATE_LIBRARIES=no $(EXTSTATIC)
 	$(Q)$(MAKE) $(EXTS_NOTE)
 
@@ -536,7 +536,7 @@ dont-install-man: $(PREP)
 post-no-install-man::
 	@$(NULLCMD)
 
-install-doc: rdoc pre-install-doc do-install-doc post-install-doc
+install-doc: pre-install-doc do-install-doc post-install-doc
 pre-install-doc:: install-prereq
 do-install-doc: $(PROGRAM) pre-install-doc
 	$(INSTRUBY) --make="$(MAKE)" $(INSTRUBY_ARGS) --install=rdoc $(INSTALL_DOC_OPTS)
@@ -1522,6 +1522,16 @@ update-man-date: PHONY
 	-e 'BEGIN{@vcs=VCS.detect(ARGV.shift)}' \
 	-e '$$_.sub!(/^(\.Dd ).*/){$$1+@vcs.modified(ARGF.path).strftime("%B %d, %Y")}' \
 	"$(srcdir)" "$(srcdir)"/man/*.1
+
+.if "sparc64" == ${MACHINE}
+cont.o: cont.c
+	@$(ECHO) compiling $<
+	$(Q) $(CC) $(CFLAGS) $(XCFLAGS) $(CPPFLAGS) -O0 $(COUTFLAG)$@ -c $<
+
+coroutine/copy/Context.o: coroutine/copy/Context.c
+	@$(ECHO) compiling $<
+	$(Q) $(CC) $(CFLAGS) $(XCFLAGS) $(CPPFLAGS) -O0 $(COUTFLAG)$@ -c $<
+.endif
 
 HELP_EXTRA_TASKS = ""
 
